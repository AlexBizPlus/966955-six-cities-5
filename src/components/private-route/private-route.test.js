import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import PrivateRoute from "./private-route";
import {Router} from "react-router-dom";
import {createBrowserHistory} from "history";
import Favorites from "../favorites/favorites";

describe(`Render PrivateRoute component`, () => {

  const browserHistory = createBrowserHistory();
  const initialStore = {
    HOTELS: {
      hotels: [{
        "city": {
          "name": `Amsterdam`
        }}],
      unsorted: [{
        "city": {
          "name": `Amsterdam`
        }}],
      offer: {
        "bedrooms": 1,
        "city": {
          "location": {
            "latitude": 1,
            "longitude": 1,
            "zoom": 1,
          },
          "name": `mock`,
        },
        "description": `mock`,
        "goods": [`1`, `2`],
        "host": {
          "avatar_url": `mock`,
          "id": 1,
          "is_pro": true,
          "name": `mock`,
        },
        "id": 1,
        "images": [`1`, `2`],
        "is_favorite": true,
        "is_premium": true,
        "location": {
          "latitude": 1,
          "longitude": 1,
          "zoom": 1,
        },
        "max_adults": 1,
        "preview_image": `mock`,
        "price": 1,
        "rating": 1,
        "title": `mock`,
        "type": `mock`,
      },
      nearby: [{}],
      update: 0,
      sort: `popular`
    },
    CITY: {
      activeCity: `Paris`,
    },
    REVIEWS: {
      reviews: [],
      reviewsCount: 0,
      review: {},
      rating: 5,
      isLoading: true,
      isError: true
    },
    MAP: {
      hover: [0, 0],
    },
    USER: {
      authorizationStatus: `AUTH`,
      login: `q@q.qq`
    }
  };
  const mockStore = configureStore([]);

  test(`Render PrivateRoute`, () => {
    const store = mockStore(initialStore);

    store.dispatch = jest.fn();
    const path = `/component`;

    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={browserHistory}>
              <PrivateRoute
                path={path}
                exact={true}
                component={Favorites}
              />
            </Router>
          </Provider>
      );

    expect(tree.root.props.children.props.children.props.component).toBe(Favorites);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
