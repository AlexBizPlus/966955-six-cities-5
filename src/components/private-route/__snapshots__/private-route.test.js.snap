// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PrivateRoute component Render PrivateRoute 1`] = `null`;
